prow/spyglass/podlogartifact_test.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestPodLogReadAll(t *testing.T) {
+	jobArtifact, err := NewPodLogArtifact("job", "123", "", 500e6, fakeJa)
+	if err != nil {
+		t.Fatalf("Pod Log ReadAll Tests failed to create pod log artifact, err %v", err)
+	}
+	jibArtifact, err := NewPodLogArtifact("jib", "123", "", 500e6, fakeJa)
+	if err != nil {
+		t.Fatalf("Pod Log ReadAll Tests failed to create pod log artifact, err %v", err)
+	}
+	testCases := []struct {
+		name     string
+		artifact *PodLogArtifact
+		expected []byte
+	}{
+		{
+			name:     "\"Job\" Podlog readall",
+			artifact: jobArtifact,
+			expected: []byte("clusterA"),
+		},
+		{
+			name:     "\"Jib\" Podlog readall",
+			artifact: jibArtifact,
+			expected: []byte("clusterB"),
+		},
+	}
+	for _, tc := range testCases {
```

Comment:
Your test cases should include handling error scenarios. Aka ensure that if a.Size() or a.fetchLogs() fail then this returns an error.  If you accept my other suggestion to switch to using an interface then you can do something like:

```
type fakePodLogJobAgent struct {
  data []bytes 
  orMaybeData map[string][]bytes
}

func (fplja *fakePodLogJobAgent, n) GetJobLogTail(name, build string) ([]byte, error) {
    if name + "/" + build != expectedName { return nil, fmt.Errorf("bad name: %v", name) }
    if build == "error" { return nil, errors.New("injected GetJobLogTail error") }
    d := fplja.data
    return d[n:len(d)-n]
}
```

Reply:
added error handling scenarios in necessary tests :+1: